### PR TITLE
Use pkg/fulcioroots and pkg/tuf from sigstore/sigstore

### PR DESCRIPTION
**Pull request by imjasonh**
Signed-off-by: Jason Hall <jason@chainguard.dev>

Depends on https://github.com/sigstore/sigstore/pull/435
Part of https://github.com/sigstore/cosign/issues/1865

#### Summary

This removes `pkg/cosign/tuf` in favor if sigstore/sigstore's new `pkg/tuf`.

This moves cosign-specific logic into an `internal` package only used by the cosign CLI. This package adds cosign-specific override behavior, and otherwise falls back to sigstore/sigstore's new `pkg/fulcioroots`.

#### Release Note

```release-note
pkg/cosign/tuf and cmd/cosign/cli/fulcio/fulcioroots packages are moved to sigstore/sigstore
```

### CODE REVIEW

File: cmd/cosign/cli/fulcio/fulcio.go [hunk 1]
```diff
@@ -187,3 +182,81 @@ func NewClient(fulcioURL string) (api.Client, error) {
...
+		var err error
+		roots, intermediates, err = initRoots()
+		if err != nil {
+			panic(err)
+		}
+	})
+	return intermediates
+}
+
+func initRoots() (*x509.CertPool, *x509.CertPool, error) {
```
Comment (haydentherapper) [inline cmd/cosign/cli/fulcio/fulcio.go:232]: Thanks for adding this here! I didn't realize this had already been added back.

**Comment by imjasonh** [comment]
Do we want to keep cosign's `pkg/tuf` around with type aliases and deprecation warnings for a release, or do we think this isn't used outside of this package? Technically I don't think we make any guarantees about the Go API stability of sigstore/cosign, though we shouldn't break people unnecessarily if we think folks are depending on it.

**Comment by dlorenc** [comment]
> Do we want to keep cosign's `pkg/tuf` around with type aliases and deprecation warnings for a release, or do we think this isn't used outside of cosign? Technically I don't think we make any guarantees about the Go API stability of sigstore/cosign, though we shouldn't break people unnecessarily if we think folks are depending on it.
> 
> I think gitsign is using it, but I can update that myself.

I don't mind removing it.

File: internal/pkg/cosign/fulcio/fulcioroots/fulcioroots.go [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+)
+
+const altRoot = "SIGSTORE_ROOT_FILE"
+
+// Get returns the Fulcio root certificate.
+//
+// If the SIGSTORE_ROOT_FILE environment variable is set, the root config found
+// there will be used instead of the normal Fulcio roots.
+//
+// If this behavior is not needed, use
```
Comment (znewman01) [inline internal/pkg/cosign/fulcio/fulcioroots/fulcioroots.go:43]: nit: not clear whether "this behavior" refers to "returns the Fulcio root certificate" or the bit about the env var.

**Review by znewman01 — DISMISSED** [review]

File: internal/pkg/cosign/fulcio/fulcioroots/fulcioroots.go (hunk 2 above)
Comment (haydentherapper) [reply internal/pkg/cosign/fulcio/fulcioroots/fulcioroots.go:43]: +1, would simply clarify that these methods support fetching from an environment variable too

File: internal/pkg/cosign/fulcio/fulcioroots/fulcioroots.go [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+)
+
+var (
+	rootsOnce        sync.Once
+	roots            *x509.CertPool
+	intermediates    *x509.CertPool
+	singletonRootErr error
+)
+
+const altRoot = "SIGSTORE_ROOT_FILE"
```
Comment (asraa) [inline internal/pkg/cosign/fulcio/fulcioroots/fulcioroots.go:36]: Move the documentation on `If the SIGSTORE_ROOT_FILE env variable is set` above to here?

Also, what is the format of this file -- notably it's a list of PEM encoded FULCIO certificates, NOT a TUF root.json.

**Comment by dlorenc** [comment]
Merge conflict, otherwise lgtm and let's get this in to unblock the log jam.

**Review by dlorenc — APPROVED** [review]

**Pull request merged**